### PR TITLE
Automate docs

### LINKED ISSUE #51: Automate docs using CI/CD
Automate docs with PR merges to `master` and new tag releases.

This issue was originally apart of #43, but was separated for ease of implementation.

Requirements
- [ ] Move documentation over to `master` from `gh-pages`
- [ ] Add publish docs job to build workflow
  - Reference [link](https://github.com/E3SM-Project/e3sm_diags/blob/fc9180f8e64bba53b3a02bf96dc887a877cf83bd/.github/workflows/build_workflow.yml#L92)
- [ ] Add publish docs job to release workflow
   - Reference [link](https://github.com/E3SM-Project/e3sm_diags/blob/fc9180f8e64bba53b3a02bf96dc887a877cf83bd/.github/workflows/release_workflow.yml#L8) 
 - [ ] QA - docs publish on fork's GitHub Pages

### PR DESCRIPTION
**Pull request by forsyth2**
Automate docs. Resolves #51.

This is technically for `v1.0.0`. However, since none of the changes should affect `zppy` users, it should be fine to merge after the release of `v1.0.0`.

### CODE REVIEW

**Comment by forsyth2** [comment]
@tomvothecoder How can I test this locally? Build the conda environment and try `make html`? How can I test the CI/CD is working?

I addressed everything on your task list in #51.

**Comment by tomvothecoder** [comment]
> @tomvothecoder How can I test this locally? Build the conda environment and try `make html`? How can I test the CI/CD is working?
> 
> I addressed everything on your task list in #51.

To test docs locally without a version dropdown, build conda env and `make html` in `docs/`.

To test docs locally with a version dropdown, you need to add `sphinx-multiversion` in the `dev.yml` and follow 6. in the [diags docs](https://e3sm-project.github.io/e3sm_diags/_build/html/master/contributing.html#edit-documentation).

To test doc builds in CI/CD, you need to:
1) Have a `gh-pages` branch on your fork
2) Have GH pages configured to point to `gh-pages`
3) Temporarily disable [line 69](https://github.com/E3SM-Project/zppy/pull/83/files#diff-febd9b6ea2b89a1713ed28b8fe8ee1ca70bce86d9ab532f2fcc4ba9920b202cbR69) of  `build-workflow.yml` so that PRs can trigger the `publish-docs` job. 
4) Check CI/CD passes and job is successful
5) Check https://forsyth2.github.io/zppy/ is built
6) Re-enable line 69

After you confirm it works, you won't need to disable and re-enable line 69 moving forward.

**Comment by tomvothecoder** [comment]
You're missing a few things to get `sphinx-multiversion`:
1) Add [`docs/source/_templates/versions.html`](https://github.com/tomvothecoder/e3sm_diags/blob/master/docs/source/_templates/versions.html). This Jinja template renders the drop down box.
2) Follow [this](https://holzhaus.github.io/sphinx-multiversion/master/quickstart.html) for adding `conf.py` settings

**Comment by tomvothecoder** [comment]
You're missing a few things to get `sphinx-multiversion` working:
1) Add [`docs/source/_templates/versions.html`](https://github.com/tomvothecoder/e3sm_diags/blob/master/docs/source/_templates/versions.html). This Jinja template renders the drop down box.
2) Follow [sphinx-multiversion quickstart](https://holzhaus.github.io/sphinx-multiversion/master/quickstart.html) for `docs/source/conf.py` settings

File: .github/workflows/build_workflow.yml [hunk 1]
```diff
@@ -64,3 +64,63 @@ jobs:
       - name: Run Unit Tests
         run: |
           python -m unittest tests/test_*.py
+
+  publish-docs:
+    #if: ${{ github.event_name == 'push' }}
```
Comment (forsyth2) [inline .github/workflows/build_workflow.yml:69]: Undo this change

File: .github/workflows/release_workflow.yml [hunk 2]
```diff
@@ -54,3 +54,62 @@ jobs:
...
+          # The below command will fail if no changes were present, so we ignore it
+          git add .
+          git commit -m "Update documentation" -a || true
+      - name: Push Changes
+        uses: ad-m/github-push-action@master
+        with:
+          branch: gh-pages
+          directory: gh-pages
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          force: true
```
Comment (forsyth2) [inline .github/workflows/release_workflow.yml:115]: Add extra line

**Comment by forsyth2** [comment]
@tomvothecoder Thanks for the guidance. I believe I've addressed all your comments. I created https://github.com/forsyth2/zppy/pull/1. The 3 checks passed on that branch before merging. I then merged but I didn't see any more checks appear. I checked that https://forsyth2.github.io/zppy/_build/html/main/index.html built successfully as well. 

I see the `publish-docs` check fails on this PR. If that's expected and everything else looks fine to you, I can merge.

**Comment by tomvothecoder** [comment]
@forsyth2 Awesome, glad to hear things went smoothly. Just a heads up, merging the PR to your fork triggered the [build workflow](https://github.com/forsyth2/zppy/actions/runs/966228530) on `main` which ran `publish-docs`.

You'll be good to go once you uncomment line #69 so that `publish-docs` is skipped for this and all subsequent PRs.
